.github/workflows/e2e.yml

```diff
@@ -0,0 +1,32 @@
+name: E2E
+
+on:
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch: { }
+
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency
+concurrency:
+  group: ${{ github.workflow }}-${{ github.actor }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+
+jobs:
+  validation:
+    name: 'Validation'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Dependencies
+        run: sudo bash ./.github/scripts/install_deps.sh
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version-file: go/src/github.com/${{ github.repository }}/go.mod
+          go-version: 1.21.3
```

Comment:
Maybe we can make a environment of go-version here?

FYI: https://github.com/xline-kv/go-xline/pull/2#discussion_r1327077970

Reply:
Done